pkg/pillar/zfs/zfs.go

```diff
@@ -151,10 +149,10 @@ func GetZvolPath(datasetName string) string {
 }
 
 // CreateVaultVolumeDataset Create an empty vault zvol
-func CreateVaultVolumeDataset(log *base.LogObject, datasetName string, zfsKeyFile string, encrypted bool, sizeBytes uint64) error {
-	// Shave off reserved + Can't align up if we're already at max space.
-	sizeBytes = sizeBytes - reserveEveStorageSizeGb - (volBlockSize * 1024)
-	alignedSize := alignUpToBlockSize(sizeBytes)
+func CreateVaultVolumeDataset(log *base.LogObject, datasetName string, zfsKeyFile string, encrypted bool, sizeBytes uint64, compressionType string, blockSize uint64) error {
+	// Shave off 1MB, Can't align up if we're already at max space.
+	sizeBytes = sizeBytes - (blockSize * 1024)
```

Comment:
What's the unit for the blockSize? If KB it might make sense to name it blockSizeKB.

Reply:
Note that some callers pass 16KBytes it seems, and here you multipy by 1024. So there is some unit confusion.

Reply:
Fixed in latest